tests/test_integration.py

```diff
@@ -14,39 +21,54 @@
 
 import docker
 
-REPO_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
-AF_ROOT = os.path.join(REPO_ROOT, 'af')
-TESTDATA_DIR = os.path.join(REPO_ROOT, 'testdata')
-
-METADB_NAME = 'ootestmetadb'
-METADB_PG_USER = 'oopguser'
-
-def start_pg(client):
-    pg_container = client.containers.run(
-            'postgres:9.6',
-            name=METADB_NAME,
-            hostname=METADB_NAME,
-            environment={
-                'POSTGRES_USER': METADB_PG_USER
-            },
-            ports={
-                '5432/tcp': 25432
-            },
-            detach=True
+REPO_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+AF_ROOT = os.path.join(REPO_ROOT, "af")
+TESTDATA_DIR = os.path.join(REPO_ROOT, "testdata")
+
+METADB_NAME = "ootestmetadb"
+METADB_PG_USER = "oopguser"
+
+
+conf = None
+
+
+def start_postgres_container(docker_client):
+    ## Start Postgres in a dedicated container named in METADB_NAME
+    running = docker_client.containers.list(filters=dict(name=METADB_NAME))
+    assert len(running) < 2
+    if len(running) == 1:
+        print("Postgres container already running: removing it")
+        running[0].stop()
+        running[0].remove()
+
+    print("Starting Postgres container")
+    pg_container = docker_client.containers.run(
+        "postgres:9.6",
+        name=METADB_NAME,
+        hostname=METADB_NAME,
+        environment={"POSTGRES_USER": METADB_PG_USER},
+        ports={"5432/tcp": 25432},
+        detach=True,
     )
     while True:
-        exit_code, output = pg_container.exec_run("psql -U {} -c 'select 1'".format(METADB_PG_USER))
+        exit_code, output = pg_container.exec_run(
+            "psql -U {} -c 'select 1'".format(METADB_PG_USER)
+        )
         if exit_code == 0:
             break
         time.sleep(0.5)
+    print("Postgres container ready")
     return pg_container
 
+
 def pg_install_tables(container):
-    _, socket = container.exec_run(cmd="psql -U {}".format(METADB_PG_USER), stdin=True, socket=True)
-    if hasattr(socket, '_sock'):
+    _, socket = container.exec_run(
+        cmd="psql -U {}".format(METADB_PG_USER), stdin=True, socket=True
+    )
+    if hasattr(socket, "_sock"):
         socket = socket._sock
```

Comment:
Why is this socket unwrapping necessary? (I think you probably didn't write this code but I am curious to know why accessing `_sock` is necessary in this context and maybe you know that.)

Reply:
Unsure. It was added in ecc43915a70027cb096465e802ba3360ceadf0a1 "Make the tests work with python2"

Reply:
I think it makes sense to document this as a py2 quirk with a comment?

Reply:
This is the relevant issue: https://github.com/docker/docker-py/issues/2255#issuecomment-475270012

Reply:
Thanks! Can we perhaps add a comment in the code mentioning the issue?

Reply:
Added a comment for this.